render/wgpu/src/context3d/current_pipeline.rs

```diff
@@ -133,24 +133,14 @@ impl CurrentPipeline {
             sampler_override: [None; 8],
         }
     }
-    pub fn set_vertex_shader(&mut self, shader: Rc<ShaderModuleAgal>) {
-        if let Some(old_shader) = &self.vertex_shader {
-            // If we change the shader, we need to recompile the pipeline.
```

Comment:
(Nit: Would it be correct to guess that you removed this because the check was/is always true, and thus `self.dirty.set(true)` should always be called?).

Reply:
I'm going to do a broader refactor of this code in follow-up pr, to perform better tracking and allow re-using a compiled shader when unrelated parts of the pipeline change. It wasn't worth it to add handling for the `None` case in this PR, only to remove it later.

Reply:
That makes good sense. Would it be good to also add a `TODO` to the code? Like:

`// TODO: Better tracking and re-usage of compiled shaders is planned to be added in a future PR.`

?

Reply:
I don't think it's really necessary